typescript/mcap/src/v0/Mcap0IndexedReader/ChunkCursor.ts

```diff
@@ -0,0 +1,245 @@
+import Heap from "heap-js";
+import { sortedIndexBy } from "lodash";
+
+import { parseRecord } from "../parse";
+import { IReadable, TypedMcapRecords } from "../types";
+
+type ChunkCursorParams = {
+  chunkIndex: TypedMcapRecords["ChunkIndex"];
+  relevantChannels: Set<number> | undefined;
+  startTime: bigint | undefined;
+  endTime: bigint | undefined;
+};
+
+/**
+ * ChunkCursor represents the reader's position in a particular chunk. The indexed reader holds
+ * ChunkCursors in a heap in order to merge multiple chunks together.
+ *
+ * Each chunk can contain multiple channels, and so a ChunkCursor itself contains a heap of cursors
+ * pointing into the message index for each channel of interest.
+ */
+export class ChunkCursor {
+  chunkIndex: TypedMcapRecords["ChunkIndex"];
```

Comment:
readonly?

Reply:
sure, but true readonly is kinda hopeless, we would need to make/import a DeepReadonly util and sprinkle it everywhere.